merge-script-assignment2.sh

```diff
@@ -1,28 +1,32 @@
-#1/bin/sh
+#!/bin/sh
 
 A2_URL=$1
 
 if [ $# -lt 1 ]
-	then
-		echo "Missing A2 URL"
-                echo "Usage: [./merge-script-assignemtn2.sh] [your_A2_Github_SSH_URL]"
+        then
+                echo "Usage: $0 [your_A2_Github_SSH_URL]"
                 exit 1
-	else
-                echo "A2 URL USED"
+        else
+                echo "Using assignment2 repo at $1"
 fi
 
+# Repoint origin to the student's assignemnt 2 repo
 git remote remove origin
-
 git remote add origin ${A2_URL}
 
-git remote add assignments-base git@github.com:cu-ecen-5013/aesd-assignments.git
+# Update tracking to the new origin
+git fetch origin
+git branch -u origin/master
 
+# Reset the assignments-base remote and fetch the latest
+git remote remove assignments-base
+git remote add assignments-base git@github.com:cu-ecen-5013/aesd-assignments.git
 git fetch assignments-base
 
+# Merge the latest assignment2 branch to current branch
 git merge -X theirs assignments-base/assignment2
-
-git checkout theirs assignments-base/assignment2
-
 git submodule update --init --recursive
 
+# Push everything up to student's assignment 2 repo
 git push origin master
```

Comment:
This seems dangerous because we could push up something that is not properly formed, and pushes to whatever the user gives in the argument. If they give assignment1 or assignment3 repo or something else, it'll happily do it and cause that repo to be set up in an unexpected way.

Reply:
Agreed, I'll add a check to see if it's the relevant assignment's URL. Thanks!